plugins/inputs/kube_inventory/pod.go

```diff
@@ -10,12 +11,20 @@ import (
 )
 
 func collectPods(ctx context.Context, acc telegraf.Accumulator, ki *KubernetesInventory) {
-	list, err := ki.client.getPods(ctx)
+	var list corev1.PodList
+	listRef := &list
+	var err error
+
+	if ki.KubeletURL != "" {
+		err = ki.LoadJSON(fmt.Sprintf("%s/pods", ki.KubeletURL), listRef)
```

Comment:
Can we please have a better function name e.g. `queryPodsFromKubelet` or similar?
```